pimd/pim_ifchannel.c

```diff
@@ -130,7 +132,7 @@ void pim_ifchannel_delete(struct pim_ifchannel *ch)
 
 	pim_ifp = ch->interface->info;
 
-	if (ch->upstream->channel_oil) {
+	if (ch->upstream && ch->upstream->channel_oil) {
```

Comment:
How can we have a ch but no upstream?

Reply:
@donaldsharp 

Before the commit https://github.com/FRRouting/frr/pull/3863 >>>>>>>
When FRR receive a PIM join for group G and no RP configured for group G, then we can have a if_channel but no upstream.

1. Create a if_channel using the function pim_ifchannel_add()
2. Update ch->upstream by calling the function pim_upstream_add()
3. Find the upstream by calling the function pim_rp_set_upstream_addr(). 
When no RP configured, it will update ch->upstream = NULL.

After the commit https://github.com/FRRouting/frr/pull/3863 >>>>>>>>
I have addressed the above issue. We cannot have a if_channel with no upstream.
1. When RP is not configured and join is received, i create a upstream with upstream address as INADDR_ANY and rpf interface as NULL.

@GrumpyOldTroll Please have a look the below commit.
https://github.com/FRRouting/frr/pull/3863 <Merged already>

Reply:
Thanks for addressing the problem.  If it stops crashing and still forwards traffic from an active adjacent active source (without an upstream pim neighbor on that interface), then I'll be satisfied and abandon my noob-hack patches in favor of this.

If either of those conditions does not hold, I would still have a problem.

I'll try to test it within 2 weeks, but I have some timed commitments that I have to work around.  Thanks again.

Reply:
OK, I tried this and it succeeds for my use case.  Thanks very much, I'm now happy with master and this pull request can be closed.